lib/posix-process/process.c

```diff
@@ -406,11 +406,13 @@ void uk_posix_process_kill(struct uk_thread *thread)
 }
 
 #if CONFIG_LIBPOSIX_PROCESS_INIT_PIDS
-static int posix_process_init(struct uk_init_ctx *ictx __unused)
+static int posix_process_init(struct uk_init_ctx *ictx)
 {
+	UK_ASSERT(ictx->tmain);
```

Comment:
Do we need to assert for this here? This function does not directly access it, it simply just passes it to `uk_posix_process_create`. I think the check for `NULL` should be offloaded to `uk_posix_process_create` instead since, unlike `posix_process_init`, it actually derefs the thread pointer/

Reply:
That depends on how you see it. One may argue that from the function's perspective it makes sense to evaluate the validity of its parameters before passing them on. Anyway I find this very minor, especially given that it's an assertion.